enhancements/network/no_overlay_mode.md

```diff
@@ -0,0 +1,807 @@
+
+## Motivation
+
+The motivations for this feature are to be found in the original upstream enhancement: https://github.com/ovn-kubernetes/ovn-kubernetes/blob/master/docs/okeps/okep-5259-no-overlay.md
+
+
+### User Stories
+See the upstream OVN-Kubernetes enhancement for detailed user stories:
+https://github.com/ovn-kubernetes/ovn-kubernetes/blob/master/docs/okeps/okep-5259-no-overlay.md#user-storiesuse-cases
+
+### Goals
+See the upstream OVN-Kubernetes enhancement for detailed goals:
+https://github.com/ovn-kubernetes/ovn-kubernetes/blob/master/docs/okeps/okep-5259-no-overlay.md#goals
+
+**Current platform limitation:**
+BGP is only supported on Bare Metal clusters. Since no-overlay mode requires BGP, it shares this limitation for now. As BGP support expands to more platforms, so will no-overlay mode.
+
+### Non-Goals
+See the upstream OVN-Kubernetes enhancement for detailed non-goals:
+https://github.com/ovn-kubernetes/ovn-kubernetes/blob/master/docs/okeps/okep-5259-no-overlay.md#non-goals
+
+## Proposal
+
+<!-- This section should explain what the proposal actually is. Enumerate -->
+<!-- *all* of the proposed changes at a *high level*, including all of the -->
+<!-- components that need to be modified and how they will be -->
+<!-- different. Include the reason for each choice in the design and -->
+<!-- implementation that is proposed here. -->
+
+<!-- To keep this section succinct, document the details like API field -->
+<!-- changes, new images, and other implementation details in the -->
+<!-- **Implementation Details** section and record the reasons for not -->
+<!-- choosing alternatives in the **Alternatives** section at the end of -->
+<!-- the document. -->
+
+The no-overlay feature largely leverages the existing BGP functionality in OVN-Kubernetes and only needs few API changes.
+The feature can be applied to:
+- the default network, at cluster installation time
+- Cluster User Defined Networks (CUDNs), at run time
+
+For each network we are going to need a `transport` parameter that takes `geneve` (default) or `no-overlay`.
+Then if `transport` is set to `no-overlay`, we need the following parameters to configure the no-overlay mode:
+- `outboundSNAT`:
+  - `Enabled`: apply source NAT to egress traffic, allowing only the node IP to be exposed, which is today's expected behaviour unless EgressIP is used;
+  - `Disabled`: do not apply any SNAT to egress traffic, thus exposing the pod subnet outside the cluster.
+- `routing`:
+  - `Managed`: delegate to OCP the configuration of the BGP routers or reflectors necessary to advertise the network pod subnet to all cluster nodes;
```

Comment:
It shall be rephrased. 

1. delegate to OVN-Kubernetes
2. OVN-K will configure the FRR instance of each node to set up an internal BGP fabric without requiring external BGP routers.